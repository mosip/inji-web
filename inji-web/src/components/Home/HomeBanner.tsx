import React from "react";
import {PlainButton} from "../Common/Buttons/PlainButton";
import { useTranslation } from "react-i18next";

export const HomeBanner: React.FC<HomeBannerProps> = (props) => {
    const {t} = useTranslation("HomePage");
    return (
        <div data-testid="HomeBanner-Container" className="py-2 pb-10">
            <div 
                data-testid="HomeBanner-Content" 
                className="mt-8 sm:mx-[4%] sm:rounded-xl pt-12 pb-6 sm:pt-16 sm:pb-8 px-[2%] sm:px-[5%] flex flex-col justify-center items-center bg-home-banner h-1/6"
            >
                <span data-testid="HomeBanner-Heading" className="text-4xl sm:text-5xl text-iw-text font-semibold text-wrap w-full sm:w-[80%] text-center pb-4">
                    {t("Banner.heading")}
                </span>
                <span data-testid="HomeBanner-Description" className="text-iw-text text-xl font-extralight w-[90%] sm:w-[75%] text-pretty text-center pb-8">
                    {t("Banner.description")}
                </span>
                <div data-testid="HomeBanner-ButtonContainer" className="w-[95%] sm:w-48">
                    <PlainButton testId="HomeBanner-Get-Started" onClick={props.onClick} title={t("Banner.buttontext")} />
                </div>
            </div>
        </div>
    );
}

export type HomeBannerProps = {
    onClick: () => void
}
