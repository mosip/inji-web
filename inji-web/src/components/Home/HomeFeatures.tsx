import React, { useState } from "react";
import { HomeFeatureItem } from "./HomeFeatureItem";
import { useTranslation } from "react-i18next";
import { IoArrowForwardCircleOutline, IoArrowBackCircleOutline } from "react-icons/io5";
import { isRTL } from "../../utils/i18n";

export const HomeFeatures: React.FC = () => {
  const { t, i18n } = useTranslation("HomePage");
  const [currentFeature, setCurrentFeature] = useState(1);
  const totalFeatures = 5;
  const rtl = isRTL(i18n.language);

  const handleNext = () => {
    setCurrentFeature((prev) => (prev + 1) % totalFeatures);
  };

  const handlePrev = () => {
    setCurrentFeature((prev) => (prev - 1 + totalFeatures) % totalFeatures);
  };

  return (
    <div data-testid="HomeFeatures-Container" className="flex justify-center items-center flex-col">
      <div data-testid="HomeFeatures-Heading" className="font-semibold text-3xl m-5">
        {t("Features.heading")}
      </div>
      <div data-testid="HomeFeatures-Description1" className="font-extralight text-center">
        {t("Features.description1")}
      </div>
      <div data-testid="HomeFeatures-Description2" className="font-extralight text-center">
        {t("Features.description2")}
      </div> 

      <img data-testid="HomeFeatures-MobileImage" className="mx-auto container scale-[80%] block sm:hidden" src={require("../../assets/InjiWebMobilePreview.png")} alt="Inji Web Mobile Preview" />
      <img data-testid="HomeFeatures-DesktopImage" className="mx-auto container scale-[80%] hidden sm:block" src={require("../../assets/InjiWebDesktopPreview.png")} alt="Inji Web Preview" />

      <div data-testid="HomeFeatures-ItemsContainer" className="flex flex-wrap gap-8 container mx-auto pb-3 md:pb-20 justify-center">
        <div data-testid="HomeFeatures-Items" className="hidden md:flex flex-wrap gap-8 justify-center">
          {[1, 2, 3, 4, 5].map((itemNo) => (
            <HomeFeatureItem key={itemNo} itemno={itemNo} />
          ))}
        </div>
        <div data-testid="HomeFeatures-MobileItem" className="sm:hidden mx-9 my-6 h-[32rem] w-full flex justify-center">
          <HomeFeatureItem itemno={currentFeature + 1} />
        </div>
      </div>
      {/* Navigation buttons and pagination dots for mobile view */}
      <div data-testid="HomeFeatures-Navigation" className="flex justify-between w-full px-5 sm:hidden items-center">
        <div data-testid="HomeFeatures-NavButtons" className="flex">
          <button onClick={handlePrev} className="bg-grey-300 p-1" aria-label="Previous feature">
            {rtl ? (
              <IoArrowForwardCircleOutline size={50} className="text-gray-500" color={"var(--iw-color-searchIcon)"} />
            ) : (
              <IoArrowBackCircleOutline size={50} className="text-gray-500" color={"var(--iw-color-searchIcon)"} />
            )}
          </button>
          <button onClick={handleNext} className="bg-grey-300 rounded mr-1" aria-label="Next feature">
            {rtl ? (
              <IoArrowBackCircleOutline size={50} className="text-gray-500" color={"var(--iw-color-searchIcon)"} />
            ) : (
              <IoArrowForwardCircleOutline size={50} className="text-gray-500" color={"var(--iw-color-searchIcon)"} />
            )}
          </button>
        </div>
        <div data-testid="HomeFeatures-Pagination" className="flex items-center px-5">
          {Array.from({ length: totalFeatures }, (_, index) => (
            <span
              key={index}
              className={`w-2 h-2 rounded-md mx-1 transition duration-300 ${
                index === currentFeature ? "scale-125 bg-gradient-to-r from-orange-500 to-purple-700 w-7 h-2 rounded-full" : "bg-gray-300"
              }`}
            ></span>
          ))}
        </div>
      </div>
    </div>
  );
};
