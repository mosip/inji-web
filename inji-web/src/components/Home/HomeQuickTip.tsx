import React from "react";
import {PlainButton} from "../Common/Buttons/PlainButton";
import { useTranslation } from "react-i18next";

export const HomeQuickTip: React.FC<HomeQuickTipProps> = (props) => {
    const { t } = useTranslation("HomePage");
    return (
        <div data-testid="HomeQuickTip-Container" className="py-2 mb-5 pb-5">
          <div data-testid="HomeQuickTip-Content" className="flex flex-col sm:flex-row items-center mx-[4%] sm:mb-3 py-[5%] sm:py-[2%] px-[1%] rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary sm:w-[91%] sm:ml-auto sm:mr-auto">
            <span data-testid="HomeQuickTip-Heading" className="w-full sm:w-1/2 text-2xl text-center sm:text-left px-[2%] text-iw-text font-semibold pb-7 sm:pb-0">
              {t("QuickTip.heading")}
            </span>
            <div data-testid="HomeQuickTip-ButtonContainer" className="w-4/5 sm:w-1/6 sm:ml-auto">
              <PlainButton testId="HomeQuickTip-Get-Started" onClick={props.onClick} title={t("QuickTip.buttontext")} />
            </div>
          </div>
        </div>
    );
};

export type HomeQuickTipProps = {
    onClick: () => void
}
