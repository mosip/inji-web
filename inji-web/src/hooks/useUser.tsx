import {useState, useEffect} from "react";
import { api } from "../utils/api";

export type User = {
    displayName: string;
    profilePictureUrl: string;
}

type ErrorObj = {
    errorCode: string;
    errorMessage: string;
}

export function useUser() {
    const [user, setUser] = useState<User | null>(null);
    const [errorObj, setErrorObj] = useState<ErrorObj | null>(null);

    useEffect(() => {
        const userData = localStorage.getItem("user");
        if (userData) {
            try {
                setUser(JSON.parse(userData));
            } catch (e) {
                console.error("Invalid user data in localStorage");
                localStorage.removeItem("user");
            }
        }
    }, []);

    const saveUser = (userData: User) => {
        localStorage.setItem("user", JSON.stringify(userData));
        setUser(userData);
    };

    const removeUser = () => {
        localStorage.removeItem("user");
        setUser(null);
    };

    const fetchUserProfile = async () => {
        try {
            const response = await fetch(api.fetchUserProfile.url(), {
                method: api.fetchUserProfile.methodType === 0 ? "GET" : "POST",
                headers: {...api.fetchUserProfile.headers()},
                credentials: "include"
            });
    
            const responseData = await response.json();
    
            if (!response.ok) {
                throw responseData;
            }
    
            const userData: User = {
                "displayName": responseData.display_name,
                "profilePictureUrl": responseData.profile_picture_url
            };

            saveUser(userData);
        } catch (error) {
            console.error("Error occurred while fetching user profile:", error);
            setUser(null);
            setErrorObj(error as ErrorObj)
            throw error;
        }
    };

    return {user, errorObj, saveUser, removeUser, fetchUserProfile};
}
