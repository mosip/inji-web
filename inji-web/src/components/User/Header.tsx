import React, {useEffect, useRef, useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {LanguageSelector} from '../Common/LanguageSelector';
import {api} from '../../utils/api';
import {toast} from 'react-toastify';
import {useUser} from '../../hooks/User/useUser';
import HamburgerMenu from '../../assets/HamburgerMenu.svg';
import {isRTL} from '../../utils/i18n';
import {useSelector} from 'react-redux';
import {RootState} from '../../types/redux';
import {useTranslation} from 'react-i18next';
import DropdownArrowIcon from '../Common/DropdownArrowIcon';
import {ROUTES} from '../../utils/constants';
import {convertStringIntoPascalCase} from "../../utils/misc";
import {navigateToUserHome} from "../../utils/navigationUtils";
import {CircleSkeleton} from '../Common/CircleSkeleton';
import {InfoFieldSkeleton} from '../Common/InfoFieldSkeleton';
import {ApiError, DropdownItem, ResponseTypeObject} from "../../types/data";
import {useApi} from "../../hooks/useApi";

type UserHeaderProps = {
    headerRef: React.RefObject<HTMLDivElement>;
    headerHeight: number;
};

export const Header: React.FC<UserHeaderProps> = ({
    headerRef,
    headerHeight
}) => {
    const language = useSelector((state: RootState) => state.common.language);
    const navigate = useNavigate();
    const [displayName, setDisplayName] = useState<string | undefined>(
        undefined
    );
    const [isProfileDropdownOpen, setIsProfileDropdownOpen] = useState(false);
    const [isHamburgerMenuOpen, setIsHamburgerMenuOpen] = useState(false);
    const dropdownRef = useRef<HTMLDivElement>(null);
    const hamburgerMenuRef = useRef<HTMLImageElement>(null);
    const {user, removeUser,isLoading} = useUser();
    const displayNameFromLocalStorage = user?.displayName;
    const hasProfilePictureUrl = user?.profilePictureUrl;
    const {t} = useTranslation('User');
    const logoutRequest = useApi()

    useEffect(() => {
        setDisplayName(displayNameFromLocalStorage);
    }, [displayNameFromLocalStorage]);

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            const target = event.target as Node;

            if (dropdownRef.current && !dropdownRef.current.contains(target)) {
                setIsProfileDropdownOpen(false);
            }

            if (
                hamburgerMenuRef.current &&
                !hamburgerMenuRef.current.contains(target)
            ) {
                setIsHamburgerMenuOpen(false);
            }
        };

        document.addEventListener('click', handleClickOutside);
        return () =>
            document.removeEventListener('click', handleClickOutside);
    }, []);

    const handleLogout = async () => {
        try {
            const response = await logoutRequest.fetchData({
                apiRequest: api.userLogout,
            })

            if (response.ok()) {
                removeUser();
                window.location.replace(ROUTES.ROOT);
            } else {
                const parsedResponse = ((response.error as ApiError)?.response?.data as ResponseTypeObject).errors
                const errorCode = parsedResponse?.[0].errorCode;
                if (errorCode === 'user_logout_error') {
                    removeUser();
                    window.location.replace(ROUTES.ROOT);
                }
                throw new Error(parsedResponse?.[0]?.errorMessage);
            }
        } catch (error) {
            console.error('Logout failed with error:', error);
            toast.error('Unable to logout. Please try again.');
        }
    };

    const dropdownItems: DropdownItem[] = [
        {
            label: t('ProfileDropdown.profile'),
            onClick: () => {
                setIsProfileDropdownOpen(false);    
                navigate(ROUTES.PROFILE,{state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-Profile'
        },
        {
            label: t('ProfileDropdown.faq'),
            onClick: () => {
                setIsProfileDropdownOpen(false);
                navigate(ROUTES.USER_FAQ, {state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-FAQ'
        },
        {
            label: t('ProfileDropdown.logout'),
            onClick: handleLogout,
            textColor: 'text-red-700',
            key: 'Profile-Dropdown-Logout'
        }
    ];

    const toggleProfileDropdown = (e: React.MouseEvent<HTMLDivElement>) => {
        e.stopPropagation(); // Prevent global click handler from immediately closing the dropdown
        setIsProfileDropdownOpen(!isProfileDropdownOpen);
    };

    const toggleHamburgerMenu = () => {
        setIsHamburgerMenuOpen(!isHamburgerMenuOpen);
    };

    const getProfileInitials = (displayName: string | undefined) => {
        return displayName ? displayName.charAt(0).toUpperCase(): 'U';
    };

    const getUserProfileIconWithName = () => {
        if (isLoading) {
            return (
              <div className="flex gap-2 items-center">
                <CircleSkeleton size="w-12 h-12" />
                <InfoFieldSkeleton width="w-24" height="h-2" />
              </div>
            );
          }
        
          return (
            <div className="flex gap-2 items-center">
              <div
                className={`aspect-square w-12 rounded-full bg-iw-avatarPlaceholder overflow-hidden flex items-center justify-center text-iw-avatarText font-medium text-lg ${
                  !hasProfilePictureUrl && 'p-2 sm:p-3 md:p-4'
                }`}
              >
                {hasProfilePictureUrl ? (
                  <img
                    src={user.profilePictureUrl}
                    alt="Profile Pic"
                    className="w-12 h-12 object-cover rounded-full"
                    referrerPolicy="no-referrer"
                  />
                ) : (
                  getProfileInitials(displayName)
                )}
              </div>
              <span className="font-semibold text-gray-800">
                {convertStringIntoPascalCase(displayName)}
              </span>
            </div>
          );
        };

    return (
        <header
            ref={headerRef}
            data-testid="dashboard-header-container"
            className="fixed top-0 left-0 right-0 z-20 bg-iw-background bg-transparent shadow-[0_4px_5px_0_rgba(0,0,0,0.051)]"
        >
            <div className="container mx-auto px-4 py-4 flex justify-between items-center">
                <div className="flex justify-start items-center gap-2">
                    <div
                        data-testid="hamburger-menu"
                        className="block sm:hidden"
                    >
                        <img
                            data-testid="icon-hamburger-menu"
                            ref={hamburgerMenuRef}
                            src={HamburgerMenu}
                            alt="Hamburger Menu"
                            onClick={toggleHamburgerMenu}
                            className="cursor-pointer"
                        />
                    </div>
                    <div
                        className="w-[130px] sm:w-[160px] md:w-[170px]"
                        role={'button'}
                        tabIndex={0}
                        onMouseDown={() => navigateToUserHome(navigate)}
                        onKeyUp={() => navigateToUserHome(navigate)}
                    >
                        <img
                            src={require('../../assets/InjiWebLogo.png')}
                            className={`max-w-full h-auto object-contain cursor-pointer ${
                                isRTL(language) ? 'mr-4' : ''
                            }`}
                            data-testid="header-injiWeb-logo"
                            alt="Inji Web Logo"
                        />
                    </div>
                </div>

                <div className="flex items-center space-x-6">
                    <LanguageSelector />

                    <div
                        data-testid="profile-details"
                        className="hidden sm:block relative"
                        ref={dropdownRef}
                    >
                        <div className="flex items-center space-x-2">
                            {getUserProfileIconWithName()}
                            {!isLoading && (
                                <div
                                className="relative inline-block cursor-pointer"
                                onClick={toggleProfileDropdown}
                                >
                                <DropdownArrowIcon isOpen={isProfileDropdownOpen} />
                                </div>
                            )}
                        </div>

                        {isProfileDropdownOpen && (
                            <div
                                data-testid="profile-dropdown"
                                className="absolute -right-7 top-100 mt-8 w-56 bg-white rounded-lg shadow-lg z-50 font-medium"
                            >
                                <div className="absolute top-[-0.3rem] right-8 w-3 h-3 bg-white transform rotate-45" />
                                <div className="py-2">
                                    {dropdownItems.map((item, index) => (
                                        <React.Fragment key={item.key}>
                                            <div
                                                className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer`}
                                                onClick={item.onClick}
                                            >
                                                {item.label}
                                            </div>
                                            {index !==
                                                dropdownItems.length - 1 && (
                                                <hr className="border-gray-200 my-1 mx-2" />
                                            )}
                                        </React.Fragment>
                                    ))}
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            </div>
            <div
                data-testid="hamburger-menu-dropdown"
                className="block sm:hidden w-full"
            >
                {isHamburgerMenuOpen && (
                    <div
                        style={{marginTop: headerHeight}}
                        className="absolute top-1 bg-white shadow-iw-hamburger-dropdown p-2 w-full"
                    >
                        <div>
                            <div className="flex items-center px-4 py-2">
                                {getUserProfileIconWithName()}
                            </div>
                            {dropdownItems.map((item, index) => (
                                <React.Fragment key={item.key}>
                                    <div
                                        className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer font-medium`}
                                        onClick={item.onClick}
                                    >
                                        {item.label}
                                    </div>
                                    {index !== dropdownItems.length - 1 && (
                                        <hr className="border-gray-200 my-1 mx-2" />
                                    )}
                                </React.Fragment>
                            ))}
                        </div>
                    </div>
                )}
            </div>
        </header>
    );
};
