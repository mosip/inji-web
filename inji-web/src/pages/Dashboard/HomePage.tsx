import React, {useState, useEffect} from "react";
import {useTranslation} from "react-i18next";
import {IssuersPage} from "../IssuersPage.tsx";
import {convertStringIntoPascalCase} from "./utils.ts";
import {useUser} from "../../hooks/useUser.tsx";

export const HomePage: React.FC = () => {
    const {t} = useTranslation("Dashboard");
    const [displayName, setDisplayName] = useState<string | undefined>(undefined);
    const {user} = useUser();
    const userDisplayName = user?.displayName;

    useEffect(() => {
        setDisplayName(userDisplayName);
    }, [userDisplayName]);

    return (
        <div className="px-4 sm:px-6 md:px-10 lg:px-20 max-w-screen-xl mx-auto my-10">
            <h1 className="text-2xl sm:text-3xl lg:text-4xl font-medium text-[#04051D] text-center mb-10">
                {`${t("Home.welcome")} ${convertStringIntoPascalCase(
                    displayName
                )}!`}
            </h1>
            <IssuersPage />
        </div>
    );
};
