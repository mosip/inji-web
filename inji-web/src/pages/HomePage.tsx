import React, {useEffect} from "react";
import {RequestStatus, useFetch} from "../hooks/useFetch";
import {IntroBox} from "../components/Common/IntroBox";
import {SearchIssuer} from "../components/Issuers/SearchIssuer";
import {IssuersList} from "../components/Issuers/IssuersList";
import {useDispatch} from "react-redux";
import {storeIssuers} from "../redux/reducers/issuersReducer";
import {api, MethodType} from "../utils/api";
import {IssuerObject} from "../types/data";
import {HeaderTile} from "../components/Common/HeaderTile";
import {useTranslation} from "react-i18next";
import {toast} from "react-toastify";

export const HomePage: React.FC = () => {

    const {state, fetchRequest} = useFetch();
    const dispatch = useDispatch();
    const {t} = useTranslation("HomePage");

    useEffect(() => {
        const fetchCall = async () => {
            const response = await fetchRequest(api.fetchIssuers(), MethodType.GET, null);
            const issuers = response?.response?.issuers.filter((issuer: IssuerObject) => issuer.credential_issuer === "Sunbird")
            dispatch(storeIssuers(issuers));
        }
        fetchCall();
    }, [])

    if (state === RequestStatus.ERROR) {
        toast.error(t("errorContent"));
    }

    return <div data-testid="Home-Page-Container">
        <div className="container mx-auto mt-8 px-4 flex-1 flex flex-col">
            <IntroBox/>
            <SearchIssuer state={state} fetchRequest={fetchRequest}/>
        </div>
        <HeaderTile content={t("containerHeading")}/>
        <IssuersList state={state}/>
    </div>

}
