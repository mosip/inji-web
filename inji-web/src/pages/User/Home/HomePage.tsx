import React, {useState, useEffect} from 'react';
import {useTranslation} from 'react-i18next';
import {IssuersPage} from '../../IssuersPage';
import {useUser} from '../../../hooks/useUser';
import {convertStringIntoPascalCase} from "../../../utils/misc";
import {HomePageStyles} from "./HomePageStyles";

export const HomePage: React.FC = () => {
    const {t} = useTranslation('Dashboard');
    const [displayName, setDisplayName] = useState<string | undefined>(undefined);
    const {user} = useUser();
    const userDisplayName = user?.displayName;

    useEffect(() => {
        setDisplayName(userDisplayName);
    }, [userDisplayName]);

    return (
        <div className={HomePageStyles.container}>
            <h1 className={HomePageStyles.welcomeText}>
                {`${t('Home.welcome')} ${convertStringIntoPascalCase(
                    displayName
                )}!`}
            </h1>
            <IssuersPage />
        </div>
    );
};
