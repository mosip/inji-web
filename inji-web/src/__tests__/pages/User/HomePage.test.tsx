import {render, screen} from '@testing-library/react';
import { useUser } from '../../../hooks/User/useUser';
import { HomePage } from '../../../pages/User/Home/HomePage';

// Mock dependencies
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (key: string) => {
            if (key === 'Home.welcome') return 'Welcome';
            return key;
        },
    }),
}));

jest.mock('../../../hooks/User/useUser', () => ({
    useUser: jest.fn(),
}));

jest.mock('../../../pages/IssuersPage', () => ({
    IssuersPage: () => <div data-testid="issuers-page">IssuersPage</div>,
}));

describe('HomePage', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('renders welcome message with PascalCase name', () => {
        (useUser as jest.Mock).mockReturnValue({
            user: {displayName: 'John Doe'},
        });

        render(<HomePage />);

        expect(screen.getByText("Welcome John Doe!")).toBeInTheDocument();
    });

    it('renders IssuersPage component', () => {
        (useUser as jest.Mock).mockReturnValue({
            user: {displayName: 'Test User'},
        });

        render(<HomePage />);
        expect(screen.getByTestId('issuers-page')).toBeInTheDocument();
    });
    
    it('matches snapshot when user is undefined', () => {
        (useUser as jest.Mock).mockReturnValue({ user: undefined });
        const { asFragment } = render(<HomePage />);
        expect(asFragment()).toMatchSnapshot();
      });
    
      it('matches snapshot when user has a displayName', () => {
        (useUser as jest.Mock).mockReturnValue({ user: { displayName: 'Jane Smith' } });
        const { asFragment } = render(<HomePage />);
        expect(asFragment()).toMatchSnapshot();
      });
});
