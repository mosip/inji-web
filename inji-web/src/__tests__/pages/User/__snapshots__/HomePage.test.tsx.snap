// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot when user has a displayName 1`] = `
<DocumentFragment>
  <div
    class="px-4 sm:px-6 md:px-10 lg:px-20 max-w-screen-xl mx-auto my-12 ml-2 sm:ml-0"
  >
    <h1
      class="text-2xl sm:text-3xl lg:text-4xl font-medium text-[#04051D] text-center mb-10"
    >
      Welcome Jane Smith!
    </h1>
    <div
      data-testid="issuers-page"
    >
      IssuersPage
    </div>
  </div>
</DocumentFragment>
`;

exports[`HomePage matches snapshot when user is undefined 1`] = `
<DocumentFragment>
  <div
    class="px-4 sm:px-6 md:px-10 lg:px-20 max-w-screen-xl mx-auto my-12 ml-2 sm:ml-0"
  >
    <h1
      class="text-2xl sm:text-3xl lg:text-4xl font-medium text-[#04051D] text-center mb-10"
    >
      Welcome undefined!
    </h1>
    <div
      data-testid="issuers-page"
    >
      IssuersPage
    </div>
  </div>
</DocumentFragment>
`;
